### ISSUE TITLE
model.score with parametrized NDCG metric

### ISSUE THREAD

**Issue by OlegBEZb** [catboost/catboost#2081]
Problem:
Training setup was the following,
```
params = {
    "iterations": 200, 
    'loss_function': 'YetiRank',
    'custom_metric': ['NDCG:top=5;type=Base;denominator=LogPosition', 'MAP:top=5'],
}  
model = CatBoostRanker(**params)  
model.fit(train_pool, eval_set=val_pool)
```
when I try to evaluate metrics on test,
```
model.score(X_test, y_test, group_id=X_test[group_col],
            top=5, type='Base', denominator='LogPosition')
```
<img width="839" alt="image" src="https://user-images.githubusercontent.com/28011503/167305885-7196bc54-25a8-4f26-8cf4-32496ae91991.png">

It looks like internally the metric is [called](https://github.com/catboost/catboost/blob/012b17550f46f999248a834ec4a7fbe8c21b0f7f/catboost/python-package/catboost/core.py#L6052) without the parameters passed either during training or during `score` call

catboost version: 1.0.5
Operating System: macOS Monterey 12.2
CPU: 2,3 GHz Quad-Core Intel Core i7
GPU: None

**Comment by andrey-khropov** [comment]
Duplicate of #2230
